src/Djot/Blocks.hs

```diff
@@ -1127,12 +1129,12 @@ closeContainingSections lev = do
       closeContainingSections lev
     _ -> pure ()
 
--- TODO avoid detour through String
+-- TODO avoid detour through Text
```

Comment:
> -- TODO avoid detour through Text

The djot spec doesn't currently mandate any particular scheme for generating auto identifiers, but this is what djot.js does:

``` js
    const base =
      s.replace(/[\]\[~!@#$%^&*(){}`,.<>\\|=+/?\s]+/g, " ")
       .trim()
       .replace(/ +/g, "-");
```
That could be done without UTF-8 encoding/decoding, all at the ByteString level.

Reply:
I wouldn't expect any of these changes to make much difference for benchmarks, though, since this function only runs once for each heading.

Reply:
Thanks, I changed it to use `Data.ByteString.Char8`.